field/internal/templates/element/temp_for_hash.go

```diff
@@ -0,0 +1,39 @@
+package element
+
+const TempForHash = `
+
+// Sgn0 is an algebraic substitute for the notion of sign in ordered fields
+// Namely, every non-zero quadratic residue in a finite field of characteristic =/= 2 has exactly two square roots, one of each sign
+// Taken from https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/ section 4.1
+// The sign of an element is not obviously related to that of its Montgomery form
+func (z *{{.ElementName}}) Sgn0() bool {
+	nonMont := *z
+	nonMont.FromMont()
+
+	return nonMont[0]%2 == 1
+}
+
+func (z *{{.ElementName}}) SetHex(hex string) {
+	var i big.Int
+	i.SetString(hex, 16)
+	if _, b := i.SetString(hex, 16); !b {
+		panic("SetString failed")
+	}
+	z.SetBigInt(&i)
+}
+
+func (z *{{.ElementName}}) EvalPolynomial(monic bool, coefficients []{{.ElementName}}, x *{{.ElementName}}) {
```

Comment:
monic boolean a part, this is duplicate with `polynomial` package. 
This API shouldn't live in the field Element package; see with @ThomasPiellard

Reply:
So just remove the monic (over-)optimization and use `polynomial`? Because as I remember that was the only reason I didn't use it.

Reply:
Yes 👍  but double check with @ThomasPiellard plan to kill the `polynomial` package. I think "Eval" method should exist somewhere.

Reply:
yes, so for this polynomial thing, either we kill the package and put Eval somewhere (in a utils file perhaps), or we can keep the package only for aliasing a Polynomial type which would be a []fr.Element, and just keep the Eval method on it.

Reply:
I'm good with both. In the meantime should I use the polynomial package?

Reply:
That is a very good candidate for generic 🙄